src/head_align_tail_dist.py

```diff
@@ -128,8 +134,11 @@ def head_align_tail(prefix, alnm_ext, mode):
                 # head and tail of a read with split alignments are considered together
                 # aligned regions of a read with split alignments are considered separately
                 # aligned regions of a circular read are considered are considered together
-
-                head_new, tail_new = get_head_tail(alnm.cigartuples)
+                if mode == "transcriptome" and (read in dict_genome_alnm_info):
+                    genome_alnm = dict_genome_alnm_info[read]
+                    head_new, tail_new = get_head_tail(genome_alnm.cigartuples)
```

Comment:
The head and tail should be the minimum observed in the sequence's transcriptome and genome alignment:
```
head_g, tail_g = get_head_tail(genome_alnm.cigartuples)
head_t, tail_t = get_head_tail(alnm.cigartuples)
head_new = min(head_g, head_t)
tail_new = min(tail_g, tail_t)
```